classes/conversion.php

```diff
@@ -316,8 +316,8 @@ private function create_conversion(\stored_file $file) : void {
         } catch (\dml_exception $e) {
             // If error is anything else but a duplicate insert, this is unexected,
             // so re-throw the error.
-            if (strpos($e->getMessage(), 'locasmarconv_pat_uix') === false
-                    && strpos($e->getMessage(), 'locasmarconv_con_uix') === false) {
+            // Postgres / Mysql error messages.
+            if (strpos($e->getMessage(), 'duplicate key value') === false || strpos($e->getMessage(), 'duplicate entry') === false) {
```

Comment:
I wonder if it has more specific error codes, or if using just `duplicate` should cover both cases without including a false positive.

Reply:
Also it seems you may have to either support the exact output as shown in the failing tests, or make the check case insensitive which might be an easier approach.

Reply:
Also consider splitting the conditionals into multiple lines. Since it's a string check it might read easier if each line is a different (string) check.